web/controllers/api/v1/message_controller.ex

```diff
@@ -6,23 +6,59 @@ defmodule Kastlex.API.V1.MessageController do
 
   plug Kastlex.Plug.EnsurePermissions
 
-  # TODO: partitioner
   def produce(conn, %{"topic" => topic, "partition" => partition} = params) do
     {partition, _} = Integer.parse(partition)
     key = Map.get(params, "key", "")
     {:ok, value, conn} = read_body(conn)
-    case :brod.produce_sync(:kastlex, topic, partition, key, value) do
+    try_produce(conn, topic, [partition], key, value, nil)
+  end
+
+  # no partition, use random partitioner
+  def produce(conn, %{"topic" => topic} = params) do
+    key = Map.get(params, "key", "")
+    {:ok, value, conn} = read_body(conn)
+    case :brod_client.get_partitions_count(:kastlex, topic) do
+      {:ok, partitions_cnt} ->
+        partitions = :lists.seq(0, partitions_cnt - 1) |> Enum.shuffle
+        try_produce(conn, topic, partitions, key, value, nil)
+      {:error, :UnknownTopicOrPartition} = error ->
```

Comment:
set of possible errors from :brod_client.get_partitions_count/2 and :brod.produce_sync/5 is different, does not make much sense to abstract error handling code for them in one common function